### DISCUSSION TITLE
input component does not support hide() show()

### DISCUSSION THREAD

**Discussion by netfaresi** [olton/Metro-UI-CSS#1913] [General]
I use MetroUI along with jQuery and jQuery loads first.

I use the input component as follows:

`<input id="inpID" type="text" data-role="input" class="mb-2 input-large" >`

I want to hide/show the rendered input field by Javascript.

But when I do the following, the input field does not disappear resp. appear:

`$('#inpID').hide()`

I used every possibility like Metro.getPlugin(el, "input")  etc. but nothing works.

Do I do a wrong thing or is this a bug?

**Comment by olton** [comment]
Yes, because a role transforms the input element into a composite element. You must apply the hide method to the parent of input
`$("#intID").closest(".input").hide()`

**Reply by netfaresi** [reply]
Many thanks.

I believe, as a "framework consumer" I naively expected to access to the show() hide() etc methods just using the ID of the original HTML element, which is the case if I use jQuery without transforming a plain html element into a composite element. 

Extending the abstraction of the handling with these methods for composite elements and integrating them into in MetroUI framework would be nice and easier to consume but it would blow up the framework size. :(

Thank you again and have a nice day!
